libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_IP.c

```diff
@@ -1824,7 +1824,7 @@ uint8_t ucProtocol;
 	{
 		return ipINVALID_LENGTH;
 	}
-	if( uxBufferLength < FreeRTOS_ntohs( pxIPPacket->xIPHeader.usLength ) )
+	if( uxBufferLength < ( size_t ) ( ipSIZE_OF_ETH_HEADER + FreeRTOS_ntohs( pxIPPacket->xIPHeader.usLength ) ) )
```

Comment:
Is the layer 2 header copied to the buffer?  Is it always an ethernet packet - could it be an 802.11 packet?

Reply:
The function `usGenerateProtocolChecksum()` expects a packet that starts with a 14-byte Ethernet header.